enhancements/sdk-generate-commands.md

```diff
@@ -0,0 +1,353 @@
+
+- \[x\] Enhancement is `implementable`
+- \[x\] Design details are appropriately documented from clear requirements
+- \[x\] Test plan is defined
+- \[ \] Graduation criteria for dev preview, tech preview, GA
+- \[ \] User-facing documentation is created.
+
+## Open Questions:
+
+* _(Old project layouts)_ Should a user be able to generate both a bundle and
+package manifests format in the same directory (for migration purposes), or should
+they be mutually exclusive, ex. `generate bundle` will return an error if a
+`generate packagemanifests` has been run in the same output directory, and visa versa?
+
+## Summary
+
+`operator-sdk generate csv` is the current entrypoint into Operator packaging
+for developers of Operator SDK projects. This command attempts to handle both
+[bundle][bundle-format] and [package manifests][package-manifests-format] format
+generation scenarios with a plethora of flags. The number of flags and confusing
+command name (`generate csv` manages higher-level formats) creates non-ideal UX.
+This proposal details new workflows for bundle and package manifests formats
+for old and [new][ep-kb-integration] `operator-sdk`-build Operator projects.
+
+To clearly define what old and new layouts mean:
+* Old project layouts: scaffolded with `operator-sdk new`
+* New project layouts: scaffolded with `operator-sdk init`
+
+We plan to improve the developer experience for operator authors using Operator SDK by providing:
+1. One `operator-sdk` subcommand _per format_ for generating manifests and metadata.
+1. Interactive-by-default commands to get UI information for packaged manifests and metadata.
+
+[bundle-format]:https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md#bundle-manifest-format
+[package-manifests-format]:https://github.com/operator-framework/operator-registry/tree/v1.5.3/#manifest-format
+[ep-kb-integration]:https://github.com/operator-framework/operator-sdk/blob/master/proposals/kubebuilder-integration.md
+
+## Motivation
+
+Currently, the `operator-sdk generate csv` command is used for generating CSVs,
+which scaffolds either a bundle or package manifests format and writes a
+[`ClusterServiceVersion` (CSV)][doc-csv] manifest to disk depending on the format selected.
+This process requires a developer to manually edit the CSV when any required
+field is missing, or if they want to add certain pieces of [UI metadata][doc-csv-metadata].
+Furthermore selecting the format is non-obvious; adding to the confusion is
+the fact that the operator-framework has no plans to deprecate the package manifests format
+in the near future, so having documentation for both formats in one command is confusing.
+
+[doc-csv]:https://github.com/operator-framework/operator-lifecycle-manager/blob/0.14.2/doc/design/building-your-csv.md
+[doc-csv-metadata]:https://github.com/operator-framework/operator-lifecycle-manager/blob/0.14.2/doc/design/building-your-csv.md#your-custom-resource-definitions
```

Comment:
Could we centralise all links at the bottom?

Reply:
I like links only used in one place localized.

Reply:
Are not all docs with the links at the end? Is not this the expected place to define the links by default?

Reply:
I can go either way. Until today I didn't know this was a markdown feature. I usually put the links directly in the paragraph.

Reply:
The motivation looks good.